eacl/table.go

```diff
@@ -248,26 +249,5 @@ func (t *Table) UnmarshalJSON(data []byte) error {
 }
 
 // EqualTables compares Table with each other.
-func EqualTables(t1, t2 Table) bool {
-	cID1, set1 := t1.CID()
-	cID2, set2 := t2.CID()
-
-	if set1 != set2 || cID1 != cID2 ||
-		!t1.Version().Equal(t2.Version()) {
-		return false
-	}
-
-	rs1, rs2 := t1.Records(), t2.Records()
-
-	if len(rs1) != len(rs2) {
-		return false
-	}
-
-	for i := 0; i < len(rs1); i++ {
-		if !equalRecords(rs1[i], rs2[i]) {
-			return false
-		}
-	}
-
-	return true
-}
+// Deprecated: compare [Table.Marshal] instead.
```

Comment:
> `waiter` does not use it just because it happened so i think

instead, i think it behaves as expected. Since specific method to exact table presence in the network, it compares its state in the Container contract

> binary representation depends on many things

false, it is stable

> this is one more pro to manual comparison, not just "are bytes equal?"

i mean its really hard to declare comparison op for such a complex type which is not `comparable` overall

> i do not agree, why?

instances may equal binary, structurally, essentially, etc. when code compares bytes explicitly, it emphasizes the criterion of equality. Again, i think waiter is doin well

---

i dont mind to keep it, but i wouldnt recommend using it in the app code, only for test purposes mb

Reply:
ofc array order will change proto binary, so equality is very context-dependent. Binary equality is what, for example, waiter expects. Otherwise, nobody should compare eACLs outside testing purposes imo

> acl table equality is a table that leads to the same access for every possible request sender

ur talkin about behavioral identity, and the current implementation violates it since a false negative return is possible: array of independent records may be shuffled w/o access changes. In this case, `EqualTables` returns false. Such tables are essentially the same, but bin diff